src/Api/CollectorApiClient.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace PrestaShop\Module\PsEventbus\Api;
+
+use GuzzleHttp\Psr7\Request;
+use PrestaShop\Module\PsEventbus\Api\Post\MultipartBody;
+use PrestaShop\Module\PsEventbus\Api\Post\PostFileApi;
+use PrestaShop\Module\PsEventbus\Config\Config;
+use Prestashop\ModuleLibGuzzleAdapter\ClientFactory;
+use Prestashop\ModuleLibGuzzleAdapter\Interfaces\HttpClientInterface;
+use PrestaShop\PsAccountsInstaller\Installer\Facade\PsAccounts;
+
+class CollectorApiClient
+{
+    /**
+     * @var HttpClientInterface
+     */
+    private $client;
+
+    /**
+     * @var \Ps_eventbus
+     */
+    private $module;
+
+    /**
+     * Accounts JSON Web token
+     *
+     * @var string
+     */
+    private $jwt;
+
+    /**
+     * @param PsAccounts $psAccounts
+     * @param string $collectorApiUrl
+     * @param \Ps_eventbus $module
+     */
+    public function __construct($psAccounts, $collectorApiUrl, $module)
+    {
+        $this->module = $module;
+        $this->jwt = $psAccounts->getPsAccountsService()->getOrRefreshToken();
+
+        // @see https://docs.guzzlephp.org/en/stable/quickstart.html
+        $this->client = (new ClientFactory())->getClient([
+            'allow_redirects' => true,
+            'base_uri' => $collectorApiUrl,
+            'connect_timeout' => 3,
+            'http_errors' => false,
+        ]);
+    }
+
+    /**
+     * Push some ShopContents to CloudSync
+     *
+     * @param string $jobId
+     * @param string $data
+     * @param int $startTime in seconds since epoch
+     * @param bool $fullSyncRequested
+     *
+     * @return array
+     */
+    public function upload(string $jobId, string $data, int $startTime, bool $fullSyncRequested = false)
```

Comment:
si tu es chaud : upload et delete me semblent identique, à l'exception de l'url...

Reply:
verb and endpoint differs, but you're right, can be slightly reduced